src/commands/server/restore.ts

```diff
@@ -85,10 +114,211 @@ export default class Restore extends Command {
+          task.title = `${task.title}OK`
+        },
+      },
+      {
+        title: 'Getting backup server info...',
+        task: async (ctx: any, task: any) => {
+          // Get all information about backup server and validate it where possible
+          // before redeploying operator to requested version.
+          ctx.backupServerConfig = getBackupServerConfiguration(flags)
+          task.title = `${task.title}OK`
+        },
+      },
+      {
+        title: 'Detecting additional parameters...',
+        task: async (ctx: any, task: any) => {
+          const kube = new KubeHelper(flags)
+          ctx.isOpenshift = await kube.isOpenShift()
+
+          // Set defaults for some parameters if they weren't set
+          if (!flags.installer) {
+            flags.installer = ctx.isOpenshift ? 'olm' : 'operator'
+          }
+          if (flags.installer === 'olm' && !flags['olm-channel']) {
+            if (await kube.operatorSubscriptionExists(SUBSCRIPTION_NAME, DEFAULT_OPENSHIFT_OPERATORS_NS_NAME)) {
+              flags['olm-channel'] = OLM_STABLE_ALL_NAMESPACES_CHANNEL_NAME
+            } else {
+              flags['olm-channel'] = OLM_STABLE_CHANNEL_NAME
+            }
+          }
+          if (ctx.isOpenshift) {
+            flags.platform = 'openshift'
+            flags['cluster-monitoring'] = true
+          } else {
+            flags.platform = 'kubernetes'
+          }
+
+          task.title = `${task.title}OK`
+        },
+      },
+      {
+        title: 'Getting installation templates...',
+        enabled: (ctx: any) => flags.version && ctx.currentOperatorVersion !== flags.version && !flags.templates,
+        task: async (ctx: any, task: any) => {
+          if (flags.installer === 'olm') {
+            // Using embedded templates here as they are used in install flow for OLM.
+            // OLM install flow should be reworked to not to use templates, but OLM dependencies.
+            flags.templates = getEmbeddedTemplatesDirectory()
+          } else {
+            const getTemplatesTasks = new Listr(undefined, ctx.listrOptions)
+            getTemplatesTasks.add(checkChectlAndCheVersionCompatibility(flags))
+            getTemplatesTasks.add(downloadTemplates(flags))
+            return getTemplatesTasks
+          }
+          task.title = `${task.title}OK`
+        },
+      },
+      {
+        title: 'Remove current Che operator',
+        enabled: (ctx: any) => ctx.isOperatorDeployed && flags.installer === 'olm' && flags.version && ctx.currentOperatorVersion !== flags.version,
+        task: async (ctx: any, _task: any) => {
+          // All preparations and validations must be done before this task!
+          // Delete old operator if any in case of OLM installer.
+          // For Operator installer, the operator deployment will be downgraded if needed.
+          const olmTasks = new OLMTasks()
+          const olmDeleteTasks = olmTasks.deleteTasks(flags)
```

Comment:
Can we avoid this?

Reply:
No, OLM does not support going back in versions.